### PR TITLE
HIVE-27729: Iceberg: Check Iceberg type in AlterTableExecuteAnalyzer

### PR DESCRIPTION
**Pull request by zhangbutao**
### What changes were proposed in this pull request?

Check Iceberg table in AlterTableExecuteAnalyzer.

### Why are the changes needed?

If we execute ROLLBACK and other cmd(expire snashot & fast_forward. etc) on a non-iceberg table, it will throw NPE. We need to check iceberg type in AlterTableExecuteAnalyzer to throw a better exception.

### Does this PR introduce _any_ user-facing change?

No
### Is the change a dependency upgrade?

No

### How was this patch tested?

UT

### CODE REVIEW

File: ql/src/java/org/apache/hadoop/hive/ql/ddl/table/execute/AlterTableExecuteAnalyzer.java [hunk 1]
```diff
@@ -69,6 +70,7 @@ public AlterTableExecuteAnalyzer(QueryState queryState) throws SemanticException
   protected void analyzeCommand(TableName tableName, Map<String, String> partitionSpec, ASTNode command)
       throws SemanticException {
     Table table = getTable(tableName);
+    DDLUtils.validateTableIsIceberg(table);
```
Comment (ayushtkn) [inline ql/src/java/org/apache/hadoop/hive/ql/ddl/table/execute/AlterTableExecuteAnalyzer.java:73]: As of today these API's are supported only for iceberg but aren't limited, shouldn't restrict or hardcode iceberg here.

**Review by ayushtkn — COMMENTED** [review]
what happens today if I try a rollback on ACID or External table? There should be some exception right?

**Comment by zhangbutao** [comment]
> what happens today if I try a rollback on ACID or External table? There should be some exception right?

Yes, rollback on ACID or External table will also has same `NPE`.

**Review by ayushtkn — COMMENTED** [review]
Quick look on the exception
It seems to be coming somewhere from here
```
      HiveStorageHandler storageHandler = createStorageHandler(table.getTTable());
      storageHandler.executeOperation(table, executeSpec);
```
is storageHandler null? Why? Can we just have a null check there & shoot an unsupported exception there, rather than these other checks?

**Review by ayushtkn — APPROVED** [review]
LGTM

**Comment by zabetak** [comment]
I accidentally referenced this PR from https://github.com/apache/hive/commit/1843105f27f0b3550f70f71e2bf44830c79a6c69. Please ignore.

**Pull request merged**